beacon-chain/blockchain/stategenerator/state_generator.go

```diff
@@ -67,12 +75,29 @@ func GenerateStateFromSlot(ctx context.Context, db *db.BeaconDB, slot uint64) (*
 		if err != nil {
 			return nil, fmt.Errorf("could not execute state transition %v", err)
 		}
+	}
 
-		root, err = hashutil.HashBeaconBlock(blk)
+	logrus.Errorf("End: Current slot %d and Finalized Epoch %d", fState.Slot, fState.FinalizedEpoch)
+
+	return fState, nil
+}
+
+func lookUpFromFinalizedBlock(db *db.BeaconDB, block *pb.BeaconBlock,
```

Comment:
Maybe a better name would be `lookUpFromHistoricalBlock` since this is not only tied with finalized block